src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/Tasks/CheckClientHandlerTypeTests.cs

```diff
@@ -45,16 +45,13 @@ public void ErrorIsNotRaised (string handler, string validBaseType)
 				ClientHandlerType = handler,
 				ResolvedAssemblies = assemblies.ToArray (),
 				ValidHandlerType = validBaseType,
+				UsingAndroidNETSdk = true,
 			};
 			Assert.True (task.Execute (), $"task should have succeeded. {string.Join (";", errors.Select (x => x.Message))}");
 		}
 
 		[Test]
-#if !NET_6
 		[TestCase ("Xamarin.Android.Net.AndroidMessageHandler", "System.Net.Http.HttpClientHandler, System.Net.Http")]
-#else
-		[TestCase ("Xamarin.Android.Net.AndroidClientHandler", "System.Net.Http.HttpMessageHandler, System.Net.Http")]
-#endif
 		public void ErrorIsRaised (string handler, string validBaseType)
```

Comment:
It looks like this `!NET_6` condition has been true (even for our .NET tests) ever since we bumped the test project to net7.0.

We should probably also produce a warning/error when `$(AndroidHttpClientHandlerType)=Xamarin.Android.Net.AndroidClientHandler`. However, the current checks in [CheckClientHandlerType](https://github.com/xamarin/xamarin-android/blob/d4c2a5e2ba148ea1d63fdc7f8c984642e7b5b369/src/Xamarin.Android.Build.Tasks/Tasks/CheckClientHandlerType.cs#L94) may not be sufficient for this due to the Xamarin.Android.Net.AndroidClientHandler : System.Net.Http.HttpClientHandler : System.Net.Http.HttpMessageHandler inheritance.

Reply:
Added https://github.com/xamarin/xamarin-android/pull/8438 to address this.